src/main/java/com/softserve/academy/spaced/repetition/service/impl/CardServiceImpl.java

```diff
@@ -82,29 +82,21 @@ public Card getCard(Long id) {
     }
 
     @Override
+    @Transactional
     public void addCard(Card card, Long deckId, List<String> imageList) {
-        if (card.getTitle().trim().isEmpty() || card.getAnswer().trim().isEmpty()
-                || card.getQuestion().trim().isEmpty()) {
-            throw new IllegalArgumentException(messageSource.getMessage("message.exception.cardFieldsNotEmpty",
-                    new Object[]{}, locale));
-        }
-        Deck deck = deckRepository.findOne(deckId);
-        card.setDeck(deck);
-        deck.getCards().add(cardRepository.save(card));
+        card.setDeck(deckRepository.findOne(deckId));
+        cardRepository.save(card);
         cardImageService.addCardImage(imageList, card);
```

Comment:
If we add some object, we should return this object (not sure)

Reply:
it returned by controller in other way